types/module/module.go

```diff
@@ -296,6 +324,11 @@ func NewManagerFromMap(moduleMap map[string]appmodule.AppModule) *Manager {
 func (m *Manager) SetOrderInitGenesis(moduleNames ...string) {
 	m.assertNoForgottenModules("SetOrderInitGenesis", moduleNames, func(moduleName string) bool {
 		module := m.Modules[moduleName]
+		if mod, ok := module.(coreAppModuleBasicAdapator); ok {
+			_, hasGenesis := mod.module.(appmodule.HasGenesis)
+			return !hasGenesis
+		}
```

Comment:
There are some of these checks that look very sketchy. Because we might have an adapted module, we need to check both if it's and adaptor and/or if it's a native core api appmodule that implements HasGenesis

Reply:
Once we move to a single manager I think we'll be able to remove the adapter

Reply:
How about let's assume that the module manager doesn't deal with the adaptor type? It's either core or legacy, but not the adaptor type and the adaptor can just be for basic manager